src/Utils/HoverEffect.vala

```diff
@@ -20,65 +20,24 @@
 */
 
 internal class HoverEffect : Clutter.Effect {
-    private class Shadow {
-        public int users;
-        public Cogl.Texture texture;
-
-        public Shadow (Cogl.Texture _texture) {
-            texture = _texture;
-            users = 1;
-        }
-    }
-
-    public int shadow_size { get; construct; }
-    public int shadow_spread { get; construct; }
-
+    public int border_size { get; construct; }
     public float scale_factor { get; set; default = 1; }
-    public uint8 shadow_opacity { get; set; default = 255; }
 
     Cogl.Material material;
 
-    public HoverEffect (int shadow_size, int shadow_spread) {
-        Object (shadow_size: shadow_size, shadow_spread: shadow_spread);
+    public HoverEffect (int border_size) {
+        Object (border_size: border_size);
     }
 
     construct {
         material = new Cogl.Material ();
     }
 
-    Cogl.Texture? get_shadow (int width, int height, int shadow_size, int shadow_spread) {
-        var buffer = new Granite.Drawing.BufferSurface (width, height);
-        buffer.context.rectangle (shadow_size - shadow_spread, shadow_size - shadow_spread,
-            width - shadow_size * 2 + shadow_spread * 2, height - shadow_size * 2 + shadow_spread * 2);
-        buffer.context.set_source_rgba (0, 130, 255, 1.0);
-        buffer.context.fill ();
-
-        var surface = new Cairo.ImageSurface (Cairo.Format.ARGB32, width, height);
-        var cr = new Cairo.Context (surface);
-
-        cr.set_source_surface (buffer.surface, 0, 0);
-        cr.paint ();
-
-        var texture = new Cogl.Texture.from_data (width, height, 0, Cogl.PixelFormat.BGRA_8888_PRE,
-            Cogl.PixelFormat.ANY, surface.get_stride (), surface.get_data ());
-
-        return texture;
-    }
-
     public override void paint (Clutter.EffectPaintFlags flags)    {
         var bounding_box = get_bounding_box ();
-        var shadow = get_shadow ((int) (bounding_box.x2 - bounding_box.x1), (int) (bounding_box.y2 - bounding_box.y1),
-            shadow_size, shadow_spread);
-
-        if (shadow != null) {
-            material.set_layer (0, shadow);
-        }
-
-        var opacity = actor.get_paint_opacity () * shadow_opacity / 255;
-        var alpha = Cogl.Color.from_4ub (255, 255, 255, opacity);
-        alpha.premultiply ();
+        var color = Cogl.Color.from_4ub (65, 201, 253, 255);
```

Comment:
Looks like this change is leaving the "ghost lines" when moving/resizing. Maybe it should be left on a separate PR? :)

Reply:
This seems to happen only with a border bigger than 1px.
I still feel that having the shadow it's a bit of an overkill. I'd like to merge this to have it this simple hover effect for now. I'll work on a separated branch just for this, to handle it better and actually trying to get the real share of the Cairo object inside the Actor.
What do you think?

Reply:
If you're going to fix it in another branch you should either make the border 1px or revert it :) just to keep master as working as possible.

It may not really be your fault though 🤔 since the border is "outside" of the actor it might be a drawing bug and we need to force a "re-draw" on that section

Reply:
Yup, I reverted to 1px! It's the lightest solution so far, and it doesn't seem to leave any ghost lines on the canvas :+1: